src/Npgsql/NpgsqlParameter.cs

```diff
@@ -510,13 +510,21 @@ internal virtual void ResolveHandler(TypeMapper typeMapper)
         if (Handler is not null)
             return;
 
-        if (_npgsqlDbType.HasValue)
-            Handler = typeMapper.ResolveByNpgsqlDbType(_npgsqlDbType.Value);
-        else if (_dataTypeName is not null)
-            Handler = typeMapper.ResolveByDataTypeName(_dataTypeName);
-        else if (_value is not null)
-            Handler = typeMapper.ResolveByValue(_value);
-        else
+        Resolve(typeMapper);
```

Comment:
Can you confirm that this is to optimize for the case where Handler is non-null (so that that fragment gets inlined)? If so, do we think that happens a lot?